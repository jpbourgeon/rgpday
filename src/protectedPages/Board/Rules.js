import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import Button from '@material-ui/core/Button'
import MobileStepper from '@material-ui/core/MobileStepper'
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft'
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight'

const styles = theme => ({
  root: {
    maxWidth: 400,
    flexGrow: 1
  }
})

class Rules extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      activeStep: 0
    }
    this.handleNext = this.handleNext.bind(this)
    this.handleBack = this.handleBack.bind(this)
    this.close = this.close.bind(this)
  }

  handleNext () {
    this.setState(prevState => ({
      activeStep: prevState.activeStep + 1
    }))
  }

  handleBack () {
    this.setState(prevState => ({
      activeStep: prevState.activeStep - 1
    }))
  }

  close () {
    this.setState({
      activeStep: 0
    }, () => {
      this.props.handleClose()
    })
  }

  render () {
    const { classes, pages } = this.props
    const { activeStep } = this.state
    const maxSteps = pages.length
    const ActiveContent = pages[activeStep]
    return (
      <Dialog
        fullWidth
        maxWidth='md'
        open={this.props.open}
        onClose={this.props.handleClose}
        aria-labelledby='max-width-dialog-title'
        scroll='paper'
      >
        <DialogContent>
          <DialogContentText component='div'>
            <ActiveContent />
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <div className={classes.navigation}>
            <MobileStepper
              steps={maxSteps}
              position='static'
              activeStep={activeStep}
              backButton={
                <Button size='small' onClick={this.handleBack} disabled={activeStep <= 0}>
                  <KeyboardArrowLeft />
                  Précédent
                </Button>
              }
              nextButton={
                <Button size='small' onClick={this.handleNext} disabled={activeStep >= maxSteps - 1}>
                  Suivant
                  <KeyboardArrowRight />
                </Button>
              }
            />
          </div>
          <Button onClick={this.close} color='primary'>
              Fermer
          </Button>
        </DialogActions>
      </Dialog>
    )
  }
}

Rules.propTypes = {
  classes: PropTypes.object.isRequired,
  pages: PropTypes.array,
  open: PropTypes.bool,
  handleOpen: PropTypes.func,
  handleClose: PropTypes.func
}

Rules.defaultPropTypes = {
  pages: [],
  open: true,
  handleOpen: () => {},
  handleClose: () => {}
}

export default withStyles(styles, { withTheme: true })(Rules)
