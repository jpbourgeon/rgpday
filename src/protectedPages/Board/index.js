import React from 'react'
import loadable from '@loadable/component'
import PropTypes from 'prop-types'
import { withStyles, withTheme } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'
import Grid from '@material-ui/core/Grid'
import Loading from 'src/components/Loading'

const styles = theme => {
  return {
    '@global': {
      '.grecaptcha-badge': {
        display: 'none !important'
      }
    },
    layout: {
      marginLeft: 'auto',
      marginRight: 'auto'
    },
    board: {
      padding: 10
    },
    svg: {
      // display: 'block'
    }
  }
}
// const AsyncScenario = loadable(props => (import(`../../scenarios/${props.scenarioId}`)))
// const AsyncBoard = loadable(props => (import(`../../scenarios/${props.scenarioId}/Board`)))

class Board extends React.Component {
  constructor (props) {
    super(props)
    this.state = {}
    this.scenarioRef = React.createRef()
    this.boardRef = React.createRef()
    this.reload = this.reload.bind(this)
  }

  componentDidMount () {
    window.addEventListener('resize', this.reload, false)
  }

  componentDidUpdate () {
    console.log(this.scenarioRef)
    if (this.scenarioRef.current) console.log(this.scenarioRef.current.getArticle(1))
  }

  componentWillUnmount () {
    window.removeEventListener('resize', this.reload, false)
  }

  reload (event) {
    event.preventDefault()
    this.forceUpdate()
  }

  render () {
    const { classes, config, theme } = this.props
    const Scenario = (config.scenarioId)
      ? loadable.lib(() => (import(`../../scenarios/${config.scenarioId}`)))
      : null
    const GameBoard = (config.scenarioId)
      ? loadable(props => (import(`../../scenarios/${config.scenarioId}/Board`)))
      : null
    const paperHeight = window.innerHeight - theme.spacing.unit * 8
    const maxSVGHeight = (window.innerWidth * 707 / 1042) - theme.spacing.unit * 8
    const height = Math.min(paperHeight, maxSVGHeight)
    return (
      <div className={classes.layout} style={{ height, width: height * 1042 / 707 }}>
        {(Scenario) ? <Scenario ref={this.scenarioRef} /> : null }
        <main>
          <Paper elevation={8} style={{ height }}>
            <Grid container>
              <Grid item xs={12} className={classes.board}>
                {(GameBoard) ? <GameBoard ref={this.boardRef} /> : <Loading />}
              </Grid>
            </Grid>
          </Paper>
        </main>
      </div>
    )
  }
}

Board.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired
}

export default withTheme()(withStyles(styles)(Board))
