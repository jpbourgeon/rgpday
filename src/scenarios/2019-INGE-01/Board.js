import React from 'react'
import { ReactComponent as BoardImage } from './board.svg'
class Board extends React.Component {
  render () {
    const { scenarioId, ...other } = this.props
    return (
      <div><BoardImage {...other} /></div>
    )
  }
}
export default Board
