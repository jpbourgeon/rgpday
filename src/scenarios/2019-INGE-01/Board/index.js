import React from 'react'
import { ReactComponent as BoardImage } from './board.svg'
import toMaterialStyle from 'material-color-hash'

class BoardComponent extends React.Component {
  constructor () {
    super()
    this.prefix = '2019-INGE-01-board-'
  }
  componentDidMount () {
    const { services, team } = this.props
    const avatarColors = toMaterialStyle(team.name || '???')
    const avatarCircle = document.getElementById(`${this.prefix}avatar-circle`)
    if (avatarCircle) avatarCircle.setAttribute('fill', avatarColors.backgroundColor)
    const avatarHtml = document.getElementById(`${this.prefix}avatar-html`)
    if (avatarHtml) {
      avatarHtml.style.color = avatarColors.color
      avatarHtml.innerHTML = team.initials
    }
    const avatarText = document.getElementById(`${this.prefix}avatar-text`)
    if (avatarText) {
      avatarText.setAttribute('color', avatarColors.color)
      avatarText.innerHTML = team.initials
    }
    Object.keys(services).forEach((id) => {
      const item = document.getElementById(this.prefix + id)
      item.style.cursor = 'pointer'
      item.style.transition = 'fill .175s ease-in'
      this.addListeners(this.props.services[id], id)
    })
  }

  componentWillUnmount () {
    const { services } = this.props
    Object.keys(services).forEach((id) => {
      this.removeListeners(this.props.services[id], id)
    })
  }

  addListeners (service, id) {
    const { pathname } = this.props
    const item = document.getElementById(this.prefix + id)
    if (service.action === 'rules') item.addEventListener('click', this.props.openRules)
    if (service.action === 'service') {
      item.addEventListener('click', () => (this.props.navigate(`${pathname}/${id}`, id)))
    }
    if (service.action === 'score') {
      item.addEventListener('click', () => (this.props.navigate(`${pathname}/score`, id)))
    }
    item.addEventListener('mouseover', () => this.fade(this.prefix + id, service.secondary))
    item.addEventListener('mouseleave', () => this.fade(this.prefix + id, service.primary))
  }

  removeListeners (service, id) {
    const { pathname } = this.props
    const item = document.getElementById(this.prefix + id)
    if (service.action === 'rules') item.removeEventListener('click', this.props.openRules)
    if (service.action === 'service') {
      item.removeEventListener('click', () => (this.props.navigate(`${pathname}/${id}`, id)))
    }
    if (service.action === 'score') {
      item.removeEventListener('click', () => (this.props.navigate(`${pathname}/score`, id)))
    }
    item.removeEventListener('mouseover', () => this.fade(this.prefix + id, service.secondary))
    item.removeEventListener('mouseleave', () => this.fade(this.prefix + id, service.primary))
  }secondary

  fade (id, color) {
    document.getElementById(id).style.fill = color
  }

  render () {
    return (
      <BoardImage />
    )
  }
}
export default BoardComponent
