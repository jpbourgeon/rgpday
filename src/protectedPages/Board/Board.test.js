/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import Board from './'
import jssSerializer from 'src/utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The Board component', () => {
  it('should render correctly', () => {
    const config = { scenarioId: '2019-INGE-01' }
    const tree = renderer.create(<Board config={config} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
