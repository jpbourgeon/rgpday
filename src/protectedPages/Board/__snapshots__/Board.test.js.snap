// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Board component should render correctly 1`] = `
<div
  className="layout-1"
  style={
    Object {
      "height": 630.7869481765835,
      "width": 929.6746817538897,
    }
  }
>
  <main>
    <div
      className="root-4 elevation8-14 rounded-5"
      style={
        Object {
          "height": 630.7869481765835,
        }
      }
    >
      <div
        className="container-31"
      >
        <div
          className="item-32 12-71 board-2"
        />
      </div>
    </div>
  </main>
</div>
`;
